thejamesgore/Foxtons-Dashboard
# feat: Add landlord dashboard frontend app

## Description 🚀

This pull request adds the landlord dashboard frontend app to the project and includes the following changes:

- Added frontend files to support the landlord dashboard app.
- Added necessary dependencies to support the frontend app, including Refine.js for data management and OAuth Google authentication for secure login.
- Updated the project's README with instructions on how to run the frontend app.

## Changes Made 👉

- Created a new directory for the landlord dashboard app under `src/pages`.
- Added new React components for the landlord dashboard app, sidebar
- Updated `src/App.js` to include routes for the landlord dashboard pages.
- Added necessary dependencies for the frontend app in `package.json`, including Refine.js 
- Configured OAuth Google authentication using Refine.js to provide secure login for users.

## Reason for Changes 🔍

By adding the landlord dashboard frontend app to our project, we can provide our users with an intuitive interface to manage their properties and tenants. By integrating OAuth Google authentication using Refine.js, we can ensure that user login is secure and user data is protected.

## How to Test

To test these changes, you can follow these steps:

- Clone this branch to your local machine.
- Install the necessary dependencies with `npm install`.
- Start the application with `npm start`.
- Verify that the application starts without any errors and that you can access the landlord dashboard pages by navigating to `http://localhost:3000/`.
- Test OAuth Google authentication by attempting to log in to the application using a Google account.

## Checklist


## Related Issues

None.

## Other Notes

None.
